### PR TITLE
Distributor validation

### LINKED ISSUE #5189: setting -distributor.ingestion-tenant-shard-size to 0 throws error
**Describe the bug**
setting -distributor.ingestion-tenant-shard-size to 0 throws error . According to the documentation setting this flag to 0 should disable the shuffle sharding. However it throws validation error that value can not be 0.

**To Reproduce**

1. Set -distributor.ingestion-tenant-shard-size to 0
2. Deploy distributors

**Expected behavior**
Documentation should be clear.

**Environment:**
 - Infrastructure: [e.g., Kubernetes, bare-metal, laptop]
 - Deployment tool: [e.g., helm, jsonnet]

**Additional Context**

### PR DESCRIPTION
**Pull request by dogukanteber**
**What this PR does**:
According [to the docs](https://cortexmetrics.io/docs/configuration/configuration-file/#limits_config), shuffle sharding should be disabled when `-distributor.ingestion-tenant-shard-size` is set to 0:

```
# The default tenant's shard size when the shuffle-sharding strategy is used.
# Must be set both on ingesters and distributors. When this setting is specified
# in the per-tenant overrides, a value of 0 disables shuffle sharding for the
# tenant.
# CLI flag: -distributor.ingestion-tenant-shard-size
[ingestion_tenant_shard_size: <int> | default = 0]
```

However, doing that returns an error. This PR adjusts the code to what the docs say.

**Which issue(s) this PR fixes**:
Fixes #5189 

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/distributor/distributor.go [hunk 1]
```diff
@@ -175,8 +174,10 @@ func (cfg *Config) Validate(limits validation.Limits) error {
 		return errInvalidShardingStrategy
 	}
 
-	if cfg.ShardingStrategy == util.ShardingStrategyShuffle && limits.IngestionTenantShardSize <= 0 {
-		return errInvalidTenantShardSize
+	if limits.IngestionTenantShardSize <= 0 {
```
Comment (friedrichg) [inline pkg/distributor/distributor.go:177]: There could be an scenario where:
1. Shuffle sharding is enabled for all tenants
2. Ingestion tenant shard is zero for all tenants
3. There is one specific tenant that has tenant shard bigger than zero
4. Shuffle sharding should be definitely enabled for that tenant. 

Is that supported with this change?

Comment (dogukanteber) [reply pkg/distributor/distributor.go:177]: I guess it does not. Let me check and make necessary changes

Comment (friedrichg) [reply pkg/distributor/distributor.go:177]: Let's make sure it is still supported

**Comment by sanyamjain22** [comment]
i think changing the validation to not throw error when shard size is specified to zero should be fine. Ring package takes care of the cases when shard size specified is <=0 or greater than number of instances. It returns the whole ring in such cases.

File: pkg/distributor/distributor.go (hunk 1 above)
Comment (dogukanteber) [reply pkg/distributor/distributor.go:178]: I checked the case you mentioned @friedrichg and made sure it is supported.

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,11 +15,11 @@
 * [ENHANCEMENT] Log: Avoid expensive log.Valuer evaluation for disallowed levels. #5297
 * [ENHANCEMENT] Improving Performance on the API Gzip Handler. #5347
 * [BUGFIX] Ruler: Validate if rule group can be safely converted back to rule group yaml from protobuf message #5265
-* [BUGFIX] Querier: Convert gRPC `ResourceExhausted` status code from store gateway to 422 limit error. #5286
```
Comment (friedrichg) [inline CHANGELOG.md:18]: we need a separate PR for this. don't fix stuff that is not related to the PR

File: pkg/distributor/distributor.go (hunk 1 above)
Comment (friedrichg) [reply pkg/distributor/distributor.go:178]: We make sure things are supported adding tests. integration or unit test

**Review by friedrichg — CHANGES_REQUESTED** [review]
